search/includes/classes/class-search.php

```diff
@@ -1258,32 +1263,49 @@ public static function ep_skip_query_integration( $skip, $query = null ) {
 	}
 
 	/**
-	 * Filter for ep_skip_query_integration that enabled rate limiting. Should be run last
+	 * Increment the query count to calculate rate-limiting
+	 * This doesn't care whether a request was successful or not.
+	 *
+	 * @param array $query formatted ElasticPress Query object
+	 * @param string $type Current query type
+	 *
+	 * @return void
+	 */
+	public function increment_rate_limit_counter_after_request( $query, $type ) {
+		if ( $type === 'query' ) {
+			self::query_count_incr();
```

Comment:
Rather than counting it here, can't we just count it inside `filter__ep_do_intercept_request()`, after the cache/not cached check, and after we make the network request to ES? We don't have to use the ES hook for this, we know when the network request completed, which is right here https://github.com/Automattic/vip-go-mu-plugins/blob/11c4384b5b38c791dab46a4426af1949dbf5d716/search/includes/classes/class-search.php#L929-L936

Reply:
I feel dumb now :)

Reply:
I feel dumb multiple times a day 😄